player/src/frame.rs

```diff
@@ -25,7 +40,17 @@ impl Frame {
     }
 
     pub fn get_data(&self) -> Arc<Vec<u8>> {
-        self.data.clone()
+        match self.data {
+            FrameData::Raw(ref data) => data.clone(),
+            _ => panic!(),
+        }
+    }
+
+    pub fn get_texture_id(&self) -> u32 {
+        match self.data {
+            FrameData::Texture(data) => data,
+            _ => panic!(),
```

Comment:
Maybe give some reason for the panic here and above, or if this is impossible to call maybe `unreachable!()` is a better choice (and it can also take a meaningful string)?